File: object_relations/app/models/restaurant.rb
class Restaurant
  attr_accessor :name

  @@restaurants = []
  @@counter = 0

  def initialize(name)
    @id = counter += 1
    @name = name
    @@restaurants << self
  end

  #find all entries on Reviews table with self's id
  #grab all corresponding customer id's
  #return list of all customer's
  def customers
    Review.all.map do |review|
      Customer.find_by_id(review.customer_id) if self.id == review.restaurant_id
    end
  end

  def reviews
    Review.all.find { |review| self.id == review.restaurant_id }
  end

  def self.find_by_id(id)
    @@restaurants.find do |restaurant|
      restaurant.id == id
    end
  end

  def self.find_by_name(name)
    @@restaurants.find do |restaurant|
      restaurant.name == name
    end
  end
end

File: javascript/models/restaurant.js
class Restaurant {
  constructor(name) {
    this.id = Store.restaurantCount++
    this.name = name
    Store.restaurants.push(this) //add Customer to Store upon initialization
    console.log(`New Restaurant - ${this.name} ${this.location}`)
  }

  findByName(name) {
    return Store.restaurants.find(
      (restaurant) => {
        return restaurant.name === name
    })
  }

  findById(id) {
    return Store.restaurants.find(
      (restaurant) => {
        return restaurant.id === id
    })
  }

  customers(){
    Store.reviews.map(
      (review) => {
        this.id === review.restaurant_id
        ? console.log(Customer.prototype.findById(review.customer_id).name)
        : ''
    })
  }

  reviews(){
    Store.reviews.map(
      (review) => {
        this.id === review.restaurant_id
        ? console.log(review)
        : ''
    })
  }
}
